### PR TITLE
feat(prettier): add prettier plugin for eslint

### PR DESCRIPTION
**Pull request by DYu24**
https://github.com/prettier/eslint-plugin-prettier
This plugin allows Prettier formatting issues to be reported as linting errors. The plugin also allows formatting to be fixed through the `lint:fix` command (`eslint --fix`), which means `npm run format` and `format:fix` will no longer be required.

### CODE REVIEW

**Comment by DYu24** [comment]
> A custom configuration should be used, that we have specified across our repositories.

By custom configuration, do you mean by the options specified in the .prettierrc.yaml found in the other repositories (and in this one as well)?

**Comment by merlinnot** [comment]
> > A custom configuration should be used, that we have specified across our repositories.
> 
> By custom configuration, do you mean by the options specified in the .prettierrc.yaml found in the other repositories (and in this one as well)?

Yes, exactly.

**Comment by DYu24** [comment]
> > > A custom configuration should be used, that we have specified across our repositories.
> > 
> > 
> > By custom configuration, do you mean by the options specified in the .prettierrc.yaml found in the other repositories (and in this one as well)?
> 
> Yes, exactly.

Right so I think how it works is that extending from `plugin:prettier/recommended` allows the eslint-plugin-prettier to be used with the options we specify in .prettierrc.yaml. On their repo, they say this:

> This does three things:
>
> * Enables eslint-plugin-prettier.
> * Sets the prettier/prettier rule to "error".
> * Extends the eslint-config-prettier configuration.
>
> You can then set Prettier's own options inside a .prettierrc file.

I've also tried testing the plugin locally, and the options specified in the .prettierrc.yaml file here does indeed come up as linting errors on the project it's applied.

**Comment by merlinnot** [comment]
> > > > A custom configuration should be used, that we have specified across our repositories.
> > > 
> > > 
> > > By custom configuration, do you mean by the options specified in the .prettierrc.yaml found in the other repositories (and in this one as well)?
> > 
> > 
> > Yes, exactly.
> 
> Right so I think how it works is that extending from `plugin:prettier/recommended` allows the eslint-plugin-prettier to be used with the options we specify in .prettierrc.yaml. On their repo, they say this:
> 
> > This does three things:
> > 
> > * Enables eslint-plugin-prettier.
> > * Sets the prettier/prettier rule to "error".
> > * Extends the eslint-config-prettier configuration.
> > 
> > You can then set Prettier's own options inside a .prettierrc file.
> 
> (https://github.com/prettier/eslint-plugin-prettier)
> 
> I've also tried testing the plugin locally, and the options specified in the .prettierrc.yaml file here does indeed come up as linting errors on the project it's applied.

The idea was to have it centralised, so there's no need to specify it in each repository separately and future changes can be made in one place and applied everywhere.

**Comment by DYu24** [comment]
> > > > > A custom configuration should be used, that we have specified across our repositories.
> > > > 
> > > > 
> > > > By custom configuration, do you mean by the options specified in the .prettierrc.yaml found in the other repositories (and in this one as well)?
> > > 
> > > 
> > > Yes, exactly.
> > 
> > 
> > Right so I think how it works is that extending from `plugin:prettier/recommended` allows the eslint-plugin-prettier to be used with the options we specify in .prettierrc.yaml. On their repo, they say this:
> > > This does three things:
> > > 
> > > * Enables eslint-plugin-prettier.
> > > * Sets the prettier/prettier rule to "error".
> > > * Extends the eslint-config-prettier configuration.
> > > 
> > > You can then set Prettier's own options inside a .prettierrc file.
> > 
> > 
> > (https://github.com/prettier/eslint-plugin-prettier)
> > I've also tried testing the plugin locally, and the options specified in the .prettierrc.yaml file here does indeed come up as linting errors on the project it's applied.
> 
> The idea was to have it centralised, so there's no need to specify it in each repository separately and future changes can be made in one place and applied everywhere.

Yes of course! The goal is to remove those configurations in those repositories once this plugin is merged as it will (and should) only rely on the configuration file in this repository. So in that sense it is centralized.

**Comment by merlinnot** [comment]
> > > > > > A custom configuration should be used, that we have specified across our repositories.
> > > > > 
> > > > > 
> > > > > By custom configuration, do you mean by the options specified in the .prettierrc.yaml found in the other repositories (and in this one as well)?
> > > > 
> > > > 
> > > > Yes, exactly.
> > > 
> > > 
> > > Right so I think how it works is that extending from `plugin:prettier/recommended` allows the eslint-plugin-prettier to be used with the options we specify in .prettierrc.yaml. On their repo, they say this:
> > > > This does three things:
> > > > 
> > > > * Enables eslint-plugin-prettier.
> > > > * Sets the prettier/prettier rule to "error".
> > > > * Extends the eslint-config-prettier configuration.
> > > > 
> > > > You can then set Prettier's own options inside a .prettierrc file.
> > > 
> > > 
> > > (https://github.com/prettier/eslint-plugin-prettier)
> > > I've also tried testing the plugin locally, and the options specified in the .prettierrc.yaml file here does indeed come up as linting errors on the project it's applied.
> > 
> > 
> > The idea was to have it centralised, so there's no need to specify it in each repository separately and future changes can be made in one place and applied everywhere.
> 
> Yes of course! The goal is to remove those configurations in those repositories once this plugin is merged as it will (and should) only rely on the configuration file in this repository. So in that sense it is centralized.

Doesn't it use the config relative to the repository which it's running against, that is if you're in `path/to/some/other/repo`, it will try to use `path/to/some/other/repo/.prettierrc`?

File: rules/prettier.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+  rules: {
+    'prettier/prettier': [
+      'error',
+      {
+        proseWrap: 'always',
+        singleQuote: true,
+        trailingComma: 'all',
```
Comment (merlinnot) [inline rules/prettier.js]: We should specify `arrowParens`: true, as it's always necessary when specifying return types in TypeScript, which we require.

We should also specify `endOfLine` as `lf`, so it's automatically fixed for Windows users.

**Pull request merged**